### PR TITLE
Update lead + lag documentation

### LINKED ISSUE #4798: Refresh lead() + lag() docs
* [ ] Proofread title, description, and parameters 
* [ ] Proofread comments in examples 
* [ ] Ensure comment headings used to divide into scannable sections 
* [ ] Ensure there's one example that shows tidy eval and links to `?dplyr_tidy_eval` 
* [ ] Ensure there's one example that shows `across()` 
* [ ] Flag examples that seem overly complex for further review 
* [ ] Switch out mtcars/iris for something more interesting, or a smaller dataset constructed specifically to illustrate one point. 
* [ ] Review size of example outputs to ensure that you can easily see important rows and cols. 
* [ ] Datasets should always be tibbles to avoid use of `as_tibble()` distracting from the main point

### PR DESCRIPTION
**Pull request by laurakwiley**
Fixes #4798

### CODE REVIEW

File: R/lead-lag.R [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
-#' lag(1:10, 1)
 #' lead(1:10, 1)
+#' lag(1:10, 1)
 #'
 #' x <- runif(5)
 #' cbind(ahead = lead(x), x, behind = lag(x))
 #'
-#' # Use order_by if data not already ordered
+#' # If data are not already ordered, use order_by
 #' df <- data.frame(year = 2000:2005, value = (0:5) ^ 2)
```
Comment (krlmlr) [inline R/lead-lag.R]: Should we use the pipe in the examples? We can use `sample_frac()` for scrambling the data.

File: R/lead-lag.R [hunk 2]
```diff
@@ -1,33 +1,33 @@
...
 #' @examples
 #' lead(1:10, 1)
 #' lead(1:10, 2)
 #'
-#' lag(1:10, 1)
 #' lead(1:10, 1)
+#' lag(1:10, 1)
 #'
 #' x <- runif(5)
 #' cbind(ahead = lead(x), x, behind = lag(x))
```
Comment (krlmlr) [inline R/lead-lag.R]: While we're at it:

```suggestion
#' tibble(ahead = lead(x), x, behind = lag(x))
```

File: R/lead-lag.R (hunk 1 above)
Comment (laurakwiley) [reply R/lead-lag.R]: Great idea! Will also make consistent with other suggestion and changing to tibble.

**Comment by hadley** [comment]
I made a couple small tweaks — the most important is that I kept the .Rd name the same to avoid breaking existing links.

**Pull request merged**